ast/ast.go

```diff
@@ -1829,6 +1833,51 @@ type CreateIndex struct {
 	InterleaveIn *InterleaveIn // optional
 }
 
+// CreateVectorIndex is CREATE VECTOR INDEX statement node.
+//
+//	CREATE VECTOR INDEX {if .IfNotExists}}IF NOT EXISTS{{end}} {{.Name | sql}}
+//	ON {{.TableName | sql}}({{.ColumnName | sql}})
+//	{{if .Where}}WHERE {{.Where | sql}}{{end}}
+//	{{.Options | sql}}
+
+type CreateVectorIndex struct {
+	// pos = Create
+	// end = Options.end
+
+	Create token.Pos // position of "CREATE" keyword
+
+	IfNotExists bool // optional
+	Name        *Ident
+	TableName   *Ident
+	ColumnName  *Ident
+	Where       *Where // optional
```

Comment:
Is this `Where` the same as a `WHERE` clause in other statements (much like `SELECT`)?

It seems to take very restricted values, so I guess it is better to introduce a new struct instead.

Reply:
Yes, it only allows `WHERE <column_name> IS NOT NULL` for now. I reuse `WHERE` clause directly for the sake of considering it may extend more WHERE conditions in the future. Then I will restrict it to `IS NOT NULL` if you prefer this way.

Reply:
Okay, I understand. Could you add a comment to mention that?

Reply:
Done, can help to take a look if it's good for you. ❤️